src/test/java/wolox/training/controllers/BaseControllerTest.java

```diff
@@ -0,0 +1,67 @@
+package wolox.training.controllers;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.ResultMatcher;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+
+public class BaseControllerTest {
+
+  @Autowired
+  protected MockMvc mockMvc;
+
+  public ResultActions deleteAndExpect(String url, ResultMatcher status, String content)
+      throws Exception {
+    return deleteWithStatus(url, status).andExpect(content().json(content));
+  }
+
+  public ResultActions getAndExpect(String url, ResultMatcher status, String content)
+      throws Exception {
+    return getWithStatus(url, status).andExpect(content().json(content));
+  }
+
+  public ResultActions deleteWithStatus(String url, ResultMatcher status) throws Exception {
+    return methodWithNoContent(delete(url), url).andExpect(status);
+  }
+
+  public ResultActions getWithStatus(String url, ResultMatcher status) throws Exception {
+    return methodWithNoContent(get(url), url).andExpect(status);
+  }
+
+  public ResultActions putWithStatus(String url, String content, ResultMatcher status)
+      throws Exception {
+    return methodWithContent(put(url), url, content).andExpect(status);
+  }
+
+  public ResultActions postWithStatus(String url, String content, ResultMatcher status)
+      throws Exception {
+    return methodWithContent(post(url), url, content).andExpect(status);
+  }
+
+  private ResultActions methodWithContent(MockHttpServletRequestBuilder method, String url,
```

Comment:
I know this method has code in common with the one below, I'm sure I could just keep separating each one in steps, but I'm not sure it's worth it. Or maybe there's a better way to test I'm not seeing.